Add some details in docs related to NFS SR settings (#502)
xcp-ng/xcp-ng-org: docs/storage/storage.md

Shared, thin-provisioned storage. Efficient, recommended for ease of maintenance and space savings.

> First, you need to create the NFS or SMB Share. There are plenty of options: dedicated hardware, dedicated software solutions (e.g. TrueNAS) or manual administration on your favorite OS. Be aware of NFS 

Reviewer: Is it relevant talking about SMB in the NFS section?

Author: Good point, this was SR ISO legacy.

Bloody duplicated code…

Author: Changed in 66e703bb142766fd8110ab5bbebd876ebde0d196

_user mapping_: to ensure correct access rights, read the provider's documentation. 

> For a _VDI NFS SR_, the share needs to be `read-write` then, on server side, adjust _export option_ especially those related to _NFS user mapping_. XO uses the `root` user to mount the share.

Reviewer: Assuming we're talking about the `/etc/exports` file:

```suggestion
For a _VDI NFS SR_, the share needs to be readable and writable. Then, on the NFS server, set up the _export table_, and the configuration options related to _user mapping_. XO uses the `root` user to mount the share.
```

> For a _VDI NFS SR_, the share needs to be `read-write` then, on server side, adjust _export option_ especially those related to _NFS user mapping_. XO uses the `root` user to mount the share.

Reviewer: It may be worth telling the user that a `exports(5)` man page exists.

Also, as a complete NFS noob, I don't really understand what there is to set up in the export table, so it may also be worth being a bit more explicit about that. 
I also don't really understand what the user mapping stuff is. Is that the "[User ID Mapping](https://man.archlinux.org/man/exports.5#User_ID_Mapping)"? If so, maybe we should also provide the configuration best practice here, if one exists.

Author: (I’m not an NFS expert either.)

> It may be worth telling the user that a `exports(5)` man page exists.

Referencing the man page is a good idea.

> I also don't really understand what the user mapping stuff is. Is that the "[User ID Mapping](https://man.archlinux.org/man/exports.5#User_ID_Mapping)"?

«User ID Mapping» is the NFS feature you mentioned; the «user mapping» I mentioned has a wider context, including user policy on the server & customer side. If you feel confused, it probably should be changed to a better expression. 

> If so, maybe we should also provide the configuration best practice here, if one exists.

I don’t think there is a «best practice». Export settings depend on the «user rights policy» on both the server and client sides. Especially: Are users identified «as themselves» on both sides? This level of knowledge is in the man page; it’s better not to reproduce it here.

Author: > Referencing the man page is a good idea.

Changed in 66e703bb142766fd8110ab5bbebd876ebde0d196

:::tip
Your host will initially mount the top-level NFS share you provide (for example, `/share/xen`). XCP-ng will then automatically create and mount a dedicated subdirectory for the SR (for example, `/share/xen/515982ab-476e-17b7-0e61-e68fef8d7d31`).

Make sure your NFS server or appliance are set to allow sub-directory mounts, or adding the SR will fail. In FreeNAS or TrueNAS, this checkbox is called "All dirs" in the NFS share properties.
:::

#### In Xen Orchestra

Go in the "New" menu entry, then Storage, and select NFS. Follow instructions from there.  

> #### In CLI
> 
> Example:
> ```shell
> xe sr-create type=nfs name-label=<SR_DESCRIPTION> shared=true device-config:server=<NFS_SERVER_IP> device-config:serverpath=</PATH/TO/SHARE> 
> ```
> You will get the SR UUID in response.

Reviewer: As I understand, this is the new way of formatting shell commands:

```suggestion
Example:

<Terminal shell title="root@xcp-ng-host — NFS">{`
xe sr-create type=nfs name-label=<SR_DESCRIPTION> shared=true device-config:server=<NFS_SERVER_IP> device-config:serverpath=</PATH/TO/SHARE>
`}</Terminal>

You will get the SR UUID in response.
```

Author: Oh… I wasn't aware of this new Docusaurus component. A rebase will fix this.

Shared, thin-provisioned storage. Efficient, recommended for ease of maintenance and space savings.

First, you need to create the NFS or SMB Share. There are plenty of options: dedicated hardware, dedicated software solutions (ie. TrueNAS) or to manual administration from your favorite OS. Be aware of NFS 

> _user mapping_ to ensure correct access rights, read the providers documentation. 

Reviewer: Also, what is meant by "correct access rights"?

```suggestion
First, you need to create the NFS or SMB Share. There are plenty of options: dedicated hardware, dedicated software solutions (e.g. TrueNAS) or manual administration on your favorite OS. Be aware of NFS 
_user mapping_: to ensure correct access rights, read the provider's documentation. 
```

> For a _VDI NFS SR_, the share needs to be `read-write` then, on server side, adjust _export option_ especially those related to _NFS user mapping_. XO use the `root` user to mount the share.

Reviewer 2: ```suggestion
For a _VDI NFS SR_, the share needs to be `read-write` then, on server side, adjust _export option_ especially those related to _NFS user mapping_. XO uses the `root` user to mount the share.
```

:::tip
Your host will mount the top-level NFS share you provide initially (ie. `/share/xen`).

> Then create folder(s) inside and mount those directly instead (ie. `/share/xen/515982ab-476e-17b7-0e61-e68fef8d7d31`)

Reviewer 3: Now that you have separated into separate sentences, this looks like you're asking users to create and mount this directory. I think you mean that it's XCP-ng which will do so automatically, and that it is why it needs to have enough rights to create directories in the share. Worth rewording.

Reviewer 3: By the way we're going from a compact TIP to one that uses much more vertical space. I'll let @thomas-dkmt decide if that's an improvement.

Reviewer 2: The multiple line breaks make the tip harder to read, in my opinion. I suggest something like that:

```
:::tip
Your host will initially mount the top-level NFS share you provide (for example, `/share/xen`). XCP-ng will then automatically create and mount a dedicated subdirectory for the SR (for example, `/share/xen/515982ab-476e-17b7-0e61-e68fef8d7d31`).

Make sure your NFS server or appliance are set to allow sub-directory mounts, or adding the SR will fail. In FreeNAS or TrueNAS, this checkbox is called "All dirs" in the NFS share properties.
:::
```

Author: Changed in ed451ac58619281d0b758982b558401fe70c27e1

This means your NFS server or appliance must be set to allow sub-directory mounts, or adding the SR will fail.

In FreeNAS or TrueNAS, this checkbox is called "All dirs" in the NFS share properties.
:::

#### In Xen Orchestra

> Go in the "New" menu entry, then Storage, and select NFS. Follow instructions from there.  

Reviewer 3: A link to the XO doc would be ideal here.

Update: but to be fair it was already missing before your PR. Consider this just a suggestion.

Author: I took a rapid look at the XO doc, I didn’t find a relevant section about our context

Shared, thin-provisioned storage. Efficient, recommended for ease of maintenance and space savings.

First, you need to create the NFS share. There are plenty of options: dedicated hardware, dedicated software solutions (e.g. TrueNAS) or manual administration on your favorite OS. Be aware of NFS user mapping: To ensure correct access rights, read the provider's documentation.

> For a VDI NFS SR, the share needs to be read-write then, on the server side, adjust the export option, especially those related to NFS user mapping. XO uses the root user to mount the share.

Reviewer 2: This sentence seems quite hard to follow, in my opinion. Is there a way to make it simpler or to split it into shorter sentences?

Author: I am not fan of the «plenty options» part. I’m not sure a reader need those information.
I kept it because «it was here», but if we agree, I’ll remove it.

Author: Suggestion: 

> First, you must create the NFS share.
> 
> Pay close attention to the NFS user mapping: to ensure correct access rights, refer to your NFS documentation.
> 
> For a NFS VDI SR, the share must be _read-write_; on the server side check the export options, particularly those related to NFS user mapping.
> 
> XO uses the root user to mount the share.

Author: We looks agree, I see you propose to remove it here:
https://github.com/xcp-ng/xcp-ng-org/pull/502/changes#r3747311412

Author: Added in 3b0e06914832f7a05b60f74a0186c6ad04f5a4e5

:::tip

> Your host will initially mount the top-level NFS share you provide (for example, `/share/xen`). XCP-ng will then automatically create and mount a dedicated subdirectory for the SR (for example, `/share/xen/515982ab-476e-17b7-0e61-e68fef8d7d31`).

Reviewer 2: ```suggestion
First, your host will mount the top-level NFS share you provide (for example, `/share/xen`). Then, XCP-ng will automatically create and mount a dedicated subdirectory for the SR (for example, `/share/xen/515982ab-476e-17b7-0e61-e68fef8d7d31`).
```

Make sure your NFS server or appliance is set to allow sub-directory mounts, or adding the SR will fail. In FreeNAS or TrueNAS, this checkbox is called "All dirs" in the NFS share properties.
:::

#### In Xen Orchestra

> Go in the "New" menu entry, then Storage, and select NFS. Follow instructions from there.  

Reviewer 2: ```suggestion
Go to the "New" menu entry, then Storage, and select NFS. Follow instructions from there.  
```

Shared, thin-provisioned storage. Efficient, recommended for ease of maintenance and space savings.

First, you need to create the NFS share. There are plenty of options: dedicated hardware, dedicated software solutions (e.g. TrueNAS) or manual administration on your favorite OS. Be aware of NFS 

> _user mapping_: to ensure correct access rights, read the provider's documentation. Reading `export(5)` man page is a good start.

Reviewer: ```suggestion
_user mapping_: to ensure correct access rights, read the provider's documentation. Reading the `export(5)` man page is a good start.
```

Shared, thin-provisioned storage. Efficient, recommended for ease of maintenance and space savings.

First, you need to create the NFS share.

Pay close attention to the NFS user mapping: to ensure correct access rights, refers to your NFS documentation.

> For a NFS VDI SR, the share must be read-write; on the server side check the export options, particularly those related to NFS user mapping.

Reviewer 2: ```suggestion
For a NFS VDI SR, the share must be read-write. On the server side, check the export options, particularly those related to NFS user mapping.
```

XO uses the root user to mount the share.

:::tip
First, your host will mount the top-level NFS share you provide (for example, `/share/xen`). Then, XCP-ng will automatically create and mount a dedicated subdirectory for the SR (for example, `/share/xen/515982ab-476e-17b7-0e61-e68fef8d7d31`).

> Make sure your NFS server or appliance is set to allow sub-directory mounts, or adding the SR will fail. In FreeNAS or TrueNAS, this checkbox is called "All dirs" in the NFS share properties.

Reviewer 2: ```suggestion
Make sure your NFS server or appliance is set to allow sub-directory mounts, or adding the SR will fail. In FreeNAS or TrueNAS, this checkbox is called **All dirs** in the NFS share properties.
```

:::

#### In Xen Orchestra

> Go to the "New" menu entry, then Storage, and select NFS. Follow instructions from there.  

Reviewer 2: ```suggestion
Go to the **New** menu entry, then **Storage**, and select **NFS**. Follow instructions from there.  
```

### Create a Shared ISO SR

First, you need to create the NFS or SMB Share. There are plenty of options: dedicated hardware, dedicated software solutions (e.g. TrueNAS) or manual administration on your favorite OS. Be aware of NFS 
_user mapping_: to ensure correct access rights, read the provider's documentation. 

> For a _ISO SR_, the share needs only to be `read-only`.

Reviewer: ```suggestion
For a _ISO SR_, the share only needs to be read-only.
```

### Create a Shared ISO SR

> First, you need to create the NFS or SMB Share. There are plenty of options: dedicated hardware, dedicated software solutions (ie. TrueNAS) or to manual administration from your favorite OS. Be aware of NFS 

Reviewer 3: Only NFS user mapping? I think SMB user mapping might matter too?

Author: I didn’t test SMB. I can do that

Author: I’ll do some SMB tests later; blocking an NFS PR for an SMB test didn’t look reasonable.

> _user mapping_ to ensure correct access rights, read the providers documentation. 

Reviewer: Ditto as above.

Author: Bloody code replication…

Changed in 3d70fec28a290fdc845d13b20e9f67b6d0aafc07

### Create a Shared ISO SR

> First, you need to create the NFS share. There are plenty of options: dedicated hardware, dedicated software solutions (e.g. TrueNAS) or manual administration on your favorite OS. Be aware of NFS 

Reviewer 2: ```suggestion
:::tip
Be aware of NFS _user mapping_: to ensure correct access rights, read the provider's documentation.
:::
```

_user mapping_: to ensure correct access rights, read the provider's documentation. 

> For an _ISO SR_, the share only needs to be read-only.

Reviewer 2: Repeating the word 'only' makes the sentence sound strange. I suggest _"For an ISO SR, the share only needs read access."_ or _"For an ISO SR, the share just needs read-only access."_, something like that.

Author: Added in fa7b7e491ec76ce5346e91ad16927751694ee239

### Create a Shared ISO SR

:::tip
Be aware of NFS _user mapping_: to ensure correct access rights, read the provider's documentation.
:::

> _user mapping_: to ensure correct access rights, read the provider's documentation. 

Reviewer 2: Is this line necessary? Seems like we're repeating the contents of line 698.

Author: You’re right 👍

> _user mapping_: to ensure correct access rights, read the provider's documentation. 

Author: ```suggestion
```
